server/text_generation_server/utils/weights.py

```diff
@@ -150,6 +150,19 @@ def get_weights_col_packed_qkv(self, prefix: str, quantize: str):
 
             bits, groupsize = self._get_gptq_params()
             weight = (qweight, qzeros, scales, g_idx, bits, groupsize, False)
+        if quantize == "awq":
+            try:
+                qweight = self._get_qweight(f"{prefix}.qweight") 
+            except RuntimeError:
+                raise RuntimeError(
+                    "Cannot load `awq` weight, make sure the model is already quantized"
+                )
+            qzeros = self._get_qweight(f"{prefix}.qzeros") 
+            scales = self._get_qweight(f"{prefix}.scales") 
+            scales = scales.to(dtype=self.dtype)
+
+            bits, groupsize = self._get_gptq_params()
+            weight = (qweight, qzeros, scales, bits, groupsize)
```

Comment:
Merging this with the `gptq` logic seems like it would be simpler to keep track of since I think everything is the same except `g_idx`

Reply:
@Narsil: This is fixed.